wordpress-mobile/WordPress-Android
# Plans on site domain update domain purchase card design

Fixes #19830 

Update site domain screen items to latest designs

| Site domain | Claim domain | Dark mode
|--------|--------|--------|
|<img src="https://github.com/wordpress-mobile/WordPress-Android/assets/990349/41b1844f-d2b1-435d-a1f4-015f957d5532" width=320>|<img src="https://github.com/wordpress-mobile/WordPress-Android/assets/2471769/8fe9061f-2b53-4fdd-95e0-f31cf84f64e5" width=320>|<img src="https://github.com/wordpress-mobile/WordPress-Android/assets/990349/0a1745c8-a496-4899-b417-98b3ccbaa53c" width=320>|

-----

## To Test:

- Launch JP app
- `Select` a site with a paid plan, no custom domain, and have admin role
- `Go` to Domains (My site -> domains or My site -> More... -> domain)
- `Verify` a Claim your domain card is displayed as in second image above
- `Tap` on Search for domain
- `Verify` it launches domain purchase flow as before
- `Try` same flow with a site that has custom domain
- `Verify` it looks like the first image above
- `Try` same flow with a site on free plan
- `Verify it displays a **Upgrade to a plan card**

-----

## Regression Notes

1. Potential unintended areas of impact

    - None

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual tests

3. What automated tests I added (or what prevented me from doing so)

    - Unit tests

-----

## PR Submission Checklist:


-----

## UI Changes Testing Checklist:
